### ISSUE TITLE
message passing

### ISSUE THREAD

**Issue by guruofquality** [guruofquality/gras#3]
Create a message passing interface that works equally in C++ and Python. This interface should (at least) be able to support everything in grextas.

Message passing will probably be wrapped over the tags interface. Perhaps the major change will be to allow the block to be "ready" for work when not all input have data available.

**Comment by guruofquality** [comment]
It looks like the post output tag is all we need for the output interface

On the input, I added erase input tags:

```
/*!
     * Erase all tags on the given input port.
     * This method may be called from the work() context
     * to erase all of the queued up tags on the input.
     * Once erased, messages cannot be propagated downstream.
     * This method allows a user to treat an input port
     * as an async message source without a data stream.
     * In this case, after processing messages from get_input_tags(),
     * the user should call erase_input_tags() before retuning from work().
     */
```

Whats missing now is setting the input reserve to 0 and giving work() an empty buffer, right now this will assert...

**Comment by guruofquality** [comment]
closed for #27
